modules/apps/29-fee/ibc_middleware.go

```diff
@@ -322,7 +322,31 @@ func (im IBCMiddleware) OnTimeoutPacket(
 
 // OnChanUpgradeInit implements the IBCModule interface
 func (im IBCMiddleware) OnChanUpgradeInit(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, sequence uint64, version, previousVersion string) (string, error) {
-	return im.app.OnChanUpgradeInit(ctx, portID, channelID, order, connectionHops, sequence, version, previousVersion)
+	var versionMetadata types.Metadata
+	if err := types.ModuleCdc.UnmarshalJSON([]byte(version), &versionMetadata); err != nil {
```

Comment:
It's currently [not allowed to initiate an upgrade with an empty `version`](https://github.com/cosmos/ibc-go/blob/3203f07f10c76d17e094773754f5da10c49933c5/modules/core/04-channel/types/upgrade.go#L54-L56), but in [`OnChanOpenInit` of 29-fee if `version` is empty, then we use the default metadata](https://github.com/cosmos/ibc-go/blob/a9c97c4024e2bd45c370e6d0829279437a0e8eee/modules/apps/29-fee/ibc_middleware.go#L49-L54) (similar thing happens in `OnChanOpenInit` of transfer). To make the open an upgrade flows more similar, would it make sense to allow empty versions and then fill them in with a default in the application callbacks?

Reply:
We discussed potentially that the empty string in an `Upgrade` may be used to signal that the version should remain unchanged. Whether we implement that or not is tbd.

I decided to leave this out of the upgrade path for fee, as I think for upgrades you should have to be explicit about the version you are providing. I discussed this async with @AdityaSripal and he agreed